src/asana/helpers.py

```diff
@@ -309,6 +309,32 @@ def urlreplace(matchobj: Match[str]) -> str:
     return re.sub(URL_REGEX, urlreplace, text)
 
 
+def get_completed_on_merge_task_ids(pull_request: PullRequest) -> List[str]:
+    """
+    Extracts task ids that the user specified should be completed on merge from
+    the body of the PR.
+    """
+    body_lines = pull_request.body().splitlines()
+    stripped_body_lines = (line.strip() for line in body_lines)
+    task_url_line = None
+    seen_asana_tasks_line = False
+    for line in stripped_body_lines:
+        if seen_asana_tasks_line:
+            task_url_line = line
+            break
+        if line.startswith("Asana tasks:"):
+            seen_asana_tasks_line = True
```

Comment:
If there's a cleaner way to do this let me know — I experimented for a bit and this was the best I could do

Reply:
I'm kinda confused - are you assuming that all `task_url`s will fit into one `line`?

Reply:
I think that each line is just all the content until a linebreak, which in markdown should be all of the task urls even if that isn't how it appears visually

Reply:
Hmm... that's definitely not what I would have guessed 🤔 I'd still expect there to be a newline character (and raw editing in GitHub would also lead me to guess that way)... but if you've manually tested this and confirmed it then I have no choice but to believe you lol

Reply:
I need to verify, that was just my assumption, so I'll take a look before merging

Reply:
Verified this works properly in https://github.com/Asana/codez/pull/102360

Reply:
@owenauch  - one weird thing I realized -- the LGTM Asana user isn't in all of our projects - if you link to a non-public task, I believe SGTM won't be able to merge the task.

Reply:
@gregeinfrank This is true for nadi as well - it won't post the PR links if the task is private, so I think this is somewhat okay

Reply:
@binoche9  -- it's a bit different though because the Nadi user (https://app.asana.com/0/438505498636/list) is actually in the asana domain and has access to all non-private tasks/projects in the Asana domain, whereas LGTM (https://app.asana.com/0/783395256970426/list) is a guest with an `@corp.asana.com` domain - and by default is _not_ a member of all projects unless explicitly added to them

Reply:
Ah TIL. Thanks for the explanation.